### PR TITLE
Mc/edit structure def settings

### PR DESCRIPTION
**Pull request by Klochette**

### CODE REVIEW

File: src/components/structuredefSettings/utils.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+export const createTitle = (
+  structureToEdit: StructureDefinition.AsObject,
+  title: string | String.AsObject
+): StructureDefinition.AsObject => {
+  if (structureToEdit.title && !title) {
+    delete structureToEdit.title;
+  } else if (
+    (!structureToEdit.title && title) ||
+    structureToEdit.title !== title
```
Comment (nriss) [inline src/components/structuredefSettings/utils.ts:17]: you can remove the condition `structureToEdit.title !== title`

File: src/components/structuredefSettings/utils.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+export const createExperimental = (
+  structureToEdit: StructureDefinition.AsObject,
+  experimental: string | Boolean.AsObject
+): StructureDefinition.AsObject => {
+  if (structureToEdit.experimental && !experimental) {
+    delete structureToEdit.experimental;
+  } else if (
+    (!structureToEdit.experimental && experimental) ||
+    structureToEdit.experimental !== experimental
```
Comment (nriss) [inline src/components/structuredefSettings/utils.ts:32]: You can remove the condition `structureToEdit.experimental !== experimental`

File: src/components/structuredefSettings/utils.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+export const createCopyright = (
+  structureToEdit: StructureDefinition.AsObject,
+  copyright: string | Markdown.AsObject
+): StructureDefinition.AsObject => {
+  if (structureToEdit.copyright && !copyright) {
+    delete structureToEdit.copyright;
+  } else if (
+    (!structureToEdit.copyright && copyright) ||
+    structureToEdit.copyright !== copyright
```
Comment (nriss) [inline src/components/structuredefSettings/utils.ts:47]: same

File: src/components/structuredefSettings/utils.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+export const createPurpose = (
+  structureToEdit: StructureDefinition.AsObject,
+  purpose: string | Markdown.AsObject | undefined
+): StructureDefinition.AsObject => {
+  if (structureToEdit.purpose && !purpose) {
+    delete structureToEdit.purpose;
+  } else if (
+    (!structureToEdit.purpose && purpose) ||
+    structureToEdit.purpose !== purpose
```
Comment (nriss) [inline src/components/structuredefSettings/utils.ts:62]: same

**Review by nriss — APPROVED** [review]

File: src/components/structuredefSettings/StructuredefSettings.tsx [hunk 5]
```diff
@@ -1,70 +1,297 @@
...
+          <option value="active">active</option>
+          <option value="draft">draft</option>
+          <option value="retired">retired</option>
+          <option value="unknown">unknown</option>
+        </select>
+        <br />
+        <label htmlFor="kind">Kind</label>
+        <select
+          name="kind"
+          onChange={(e) => editKind(e)}
```
Comment (nriss) [inline src/components/structuredefSettings/StructuredefSettings.tsx:259]: same

**Review by nriss — CHANGES_REQUESTED** [review]

File: src/components/structuredefSettings/StructuredefSettings.tsx [hunk 6]
```diff
@@ -1,70 +1,297 @@
...
+    <>
+      <h2>Structure definition Settings</h2>
+      <form action="">
+        <label htmlFor="name">Name of profile</label>
+        <input
+          type="text"
+          name="name"
+          value={(name as unknown) as string}
+          onChange={(e) => editName(e)}
+        />
```
Comment (BPierrick) [inline src/components/structuredefSettings/StructuredefSettings.tsx:199]: This HTML tag couple seems to be very redundant.
Maybe creating a generic react component would be useful in this situation.

Also that way, when you'll start to apply style in your components, you would only have to set it in your custom Input label.

**Review by nriss — APPROVED** [review]

**Pull request merged**